livekit_agent.proto

```diff
@@ -74,38 +80,72 @@ enum JobStatus {
     JS_FAILED = 2;
 }
 
+message SimulateJobRequest {
+    JobType type = 1;
+    string room_id = 2;
+    string participant_id = 3;
+}
+
+message WorkerPing {
+    int64 timestamp = 1;
+}
+message WorkerPong {
+    int64 last_timestamp = 1;
+    int64 timestamp = 2;
+}
+
 message RegisterWorkerRequest {
     JobType type = 1;
     string worker_id = 2;
     string version = 3;
     string name = 4;
+    uint32 ping_interval = 5;
+    optional string namespace = 6;
 }
 
 message RegisterWorkerResponse {
     string worker_id = 1;
     string server_version = 2;
 }
 
+message MigrateJobRequest {
+    Job job = 1;
+}
+
 message AvailabilityRequest {
     Job job = 1;
+
+    // True when the job was previously assigned to another worker but has been 
+    // migrated due to different reasons (e.g. worker failure, job migration)
+    bool resuming = 2;
 }
 
 message AvailabilityResponse {
     string job_id = 1;
     bool available = 2;
+    bool should_retry = 3;
+
+    // How much the worker thinks the job will cost in terms of resources
+    double job_cost = 4;
```

Comment:
would this be required?

Reply:
After receiving a list of worker that can handle this job, the job_cost could help to pick the worker we want to chose. This may be useful for users that want that prefer to accept a job on a specific worker.
So this is indeed optional. This is similar of what psrpc is doing with "affinity" I think.
Wdyt? The naming here may not be very good

Reply:
if the worker has already accepted the job, then it would mean that it has capacity for it (it should be up to the worker to decide)

psrpc affinity is using current load metrics to decide.